src/lib/pgp-key.cpp

```diff
@@ -1439,27 +1453,15 @@ pgp_key_t::valid_till_common(bool expiry) const
 uint64_t
 pgp_key_t::valid_till() const
 {
-    if (!is_primary()) {
-        RNP_LOG("must be called for primary key only");
-        throw rnp::rnp_exception(RNP_ERROR_BAD_STATE);
-    }
-    return valid_till_common(expired());
+    return valid_till_;
 }
 
-uint64_t
-pgp_key_t::valid_till(const pgp_key_t &primary) const
+bool
+pgp_key_t::valid_at(uint64_t timestamp) const
 {
-    if (!is_subkey()) {
-        RNP_LOG("must be called for subkey only");
-        throw rnp::rnp_exception(RNP_ERROR_BAD_STATE);
-    }
-    uint64_t till = primary.valid_till();
-    /* if primary key was never valid then subkey was not either */
-    if (!till) {
-        return till;
-    }
-    /* subkey cannot be valid longer then the primary key */
-    return std::min(till, valid_till_common(expired() || primary.expired()));
+    /* TODO: consider implementing more sophisticated checks, as key validity time could
+     * possibly be non-continuous */
+    return (timestamp >= creation()) && timestamp && (timestamp <= valid_till());
```

Comment:
Have we considered y2k38 problem here? That is, MAX_INT32 != MAX_INT32 on some systems.
However, this method is probably used for validity of creation() done in the past, and people will hardly use Debian9 after year 2038....

Reply:
@rrrooommmaaa All variables here are `uint64_t`, or do I miss something?

Reply:
Here -- yes. I'm thinking how they got here -- probably extended from int32_t, where MAX_INT32 has special meaning on Debian9. Just thinking.